From 9890f46e6441b24515db784c1bcbc04486e0f56c Mon Sep 17 00:00:00 2001
From: ShrykeWindgrace <shryke.windgrace@gmail.com>
Date: Fri, 8 Dec 2023 15:18:16 +0100
Subject: [PATCH 2/2] feat: support hledger-ui on windows

Introduced or bumped dependencies:

- brick-2.1.1
- vty-6.1
- vty-crossplatform-0.4.0.0
- vty-windows-0.2.0.1 conditionally on windows (current version of
  vty-crossplatform has 0.2.0.0 as a lower bound, need to put a newer version explcitly;
  once we get a newer vty-crossplatform, we will be able to drop this conditional)
- vty-unix (indirectly via vty-crossplatform)
---
 hledger-ui/Hledger/UI/Main.hs |  3 ++-
 hledger-ui/hledger-ui.cabal   | 16 +++++++---------
 hledger-ui/package.yaml       | 25 ++++++++++++++-----------
 stack.yaml                    |  7 +++++++
 stack9.4.yaml                 |  8 ++++++--
 5 files changed, 36 insertions(+), 23 deletions(-)

diff --git a/hledger-ui/Hledger/UI/Main.hs b/hledger-ui/Hledger/UI/Main.hs
index 8dd9d7f369f..628815cc6f2 100644
--- a/hledger-ui/Hledger/UI/Main.hs
+++ b/hledger-ui/Hledger/UI/Main.hs
@@ -20,7 +20,8 @@ import Data.List (find)
 import Data.List.Extra (nubSort)
 import Data.Maybe (fromMaybe)
 import qualified Data.Text as T
-import Graphics.Vty (mkVty, Mode (Mouse), Vty (outputIface), Output (setMode))
+import Graphics.Vty (Mode (Mouse), Vty (outputIface), Output (setMode))
+import Graphics.Vty.CrossPlatform (mkVty)
 import Lens.Micro ((^.))
 import System.Directory (canonicalizePath)
 import System.Environment (withProgName)
diff --git a/hledger-ui/hledger-ui.cabal b/hledger-ui/hledger-ui.cabal
index 30b46e597fd..7bf5c958b98 100644
--- a/hledger-ui/hledger-ui.cabal
+++ b/hledger-ui/hledger-ui.cabal
@@ -73,7 +73,7 @@ library
       ansi-terminal >=0.9
     , async
     , base >=4.14 && <4.19
-    , brick >=1.5 && <2
+    , brick >=2.0 && <2.2
     , cmdargs >=0.8
     , containers >=0.5.9
     , data-default
@@ -95,14 +95,16 @@ library
     , text-zipper >=0.4
     , time >=1.5
     , transformers
-    , unix
     , vector
-    , vty >=5.15 && <6
+    , vty >=6.1
+    , vty-crossplatform >=0.4.0.0 && <0.5.0.0
   default-language: Haskell2010
   if os(windows)
-    buildable: False
+    build-depends:
+        vty-windows >=0.2.0.1 && <0.3.0.0
   else
-    buildable: True
+    build-depends:
+        unix
 
 executable hledger-ui
   main-is: hledger-ui.hs
